dodiscover/context.py

```diff
@@ -112,3 +120,21 @@ def excluded_edges(self) -> nx.Graph:
     @property
     def init_graph(self) -> GraphProtocol:
         return self._init_graph
+
+    def copy(self):
```

Comment:
I expect, given our current planning for context, that a copy method on a context object should produce [deep copies](https://docs.python.org/3/library/copy.html#copy.deepcopy). ping @darthtrevino

Reply:
Changed to deepcopy. Yeah this is a prototype just to demonstrate how it would work. Most likely will be superseded by the API in #42